packages/stats/src/index.ts

```diff
@@ -0,0 +1,160 @@
+import * as Parser from '@tracespace/parser'
+
+export interface DrillUsage {
+  diameter: number
+  count: number
+}
+
+export interface DrillStats {
+  drillHits: DrillUsage[]
+  drillRoutes: DrillUsage[]
+  totalDrills: number
+  totalRoutes: number
+  minDrillSize: number
+  maxDrillSize: number
+}
+
+/**
+ * Method to collect drill stats of parsed drill files.
+ *
+ * @example
+ * ```ts
+ * import {createParser} from '@tracespace/parser'
+ * import {collectDrillStats} from '@tracespace/stats'
+ *
+ * const parser = createParser()
+ *
+ * parser.feed(...)
+ *
+ * const tree = parser.result()
+ * const stats = collectDrillStats([tree])
+ * ```
+ *
+ * @category Stats
+ */
+export function collectDrillStats(trees: Parser.Root[]): DrillStats {
+  const state: DrillStatsState = {
+    usedTools: {},
+    drillsPerTool: {},
+    routesPerTool: {},
+    totalDrills: 0,
+    totalRoutes: 0,
+    minDrillSize: null,
+    maxDrillSize: null,
+  }
+
+  for (const tree of trees) {
+    if (tree.filetype !== Parser.DRILL) {
+      continue
+    }
+
+    _updateDrillStats(state, tree)
+  }
+
+  const hits = Object.entries(state.drillsPerTool).map(([key, count]) => ({
+    diameter: state.usedTools[key],
+    count,
+  }))
+  const routes = Object.entries(state.routesPerTool).map(([key, count]) => ({
+    diameter: state.usedTools[key],
+    count,
+  }))
+
+  const stats: DrillStats = {
+    drillHits: hits,
+    drillRoutes: routes,
+    totalDrills: state.totalDrills,
+    totalRoutes: state.totalRoutes,
+    minDrillSize: state.minDrillSize ?? 0,
+    maxDrillSize: state.maxDrillSize ?? 0,
+  }
+
+  return stats
+}
+
+interface DrillStatsState {
+  usedTools: Record<string, number>
+  drillsPerTool: Record<string, number>
+  routesPerTool: Record<string, number>
+  totalDrills: number
+  totalRoutes: number
+  minDrillSize: number | null
+  maxDrillSize: number | null
+}
+
+function _updateDrillStats(state: DrillStatsState, tree: Parser.Root) {
+  let currentTool = null
+  let currentMode: Parser.InterpolateModeType = Parser.DRILL
+  for (const node of tree.children) {
+    switch (node.type) {
+      case Parser.TOOL_DEFINITION:
+        if (node.shape.type !== Parser.CIRCLE) {
+          continue
+        }
+
+        currentTool = node.code
+
+        state.usedTools[currentTool] = node.shape.diameter
+        break
+      case Parser.TOOL_CHANGE:
+        currentTool = node.code
+        break
+      case Parser.INTERPOLATE_MODE:
+        currentMode = node.mode
+        break
+      case Parser.GRAPHIC: {
+        if (currentTool === null) {
+          continue
+        }
+
+        const diameter = state.usedTools[currentTool]
+        state.minDrillSize = _cmp(state.minDrillSize, diameter, Math.min)
+        state.maxDrillSize = _cmp(state.maxDrillSize, diameter, Math.max)
```

Comment:
I would prefer calling `Math.min` and `Math.max` directly over the `_cmp` misdirection

```suggestion
        state.minDrillSize = Math.min(
          diameter,
          state.minDrillSize ?? Number.POSITIVE_INFINITY
        )
        state.maxDrillSize = Math.max(
          diameter,
          state.maxDrillSize ?? Number.NEGATIVE_INFINITY
        )
```